javascript_nodejs/14.nlp-with-dispatch/dialogs/homeAutomation/index.js

```diff
@@ -62,22 +62,26 @@ class homeAutomation {
                 break;
             case NONE_INTENT:
             default:
-                await context.sendActivity(`HomeAutomation dialog cannot fulfill this request. Bubbling up`);
-                // TODO: this dialog cannot handle this specific utterance. bubble up to parent
+                await context.sendActivity(`HomeAutomation dialog cannot fulfill this request.`);
         }
-        
     }
     /**
      * 
      * @param {Object} homeAutoResults results from LUIS recognizer
      * @param {Object} context context object
      */
     async handleDeviceUpdate(homeAutoResults, context) {
+        // Find entities from the LUIS model
+        // The LUIS language understnading model is set up with both machine learned simple entities as well as pattern.any entities. 
+        // We should look to see if either one have value. In cases where both have value, we will take the machine learned simple entities value.
         const devices = findEntities(DEVICE_ENTITY, homeAutoResults.entities);
         const devices_patternAny = findEntities(DEVICE_PATTERNANY_ENTITY, homeAutoResults.entities);
+        // Find any rooms specified in user uttrance. Look for both machine learned simple entity as well as pattern.any entity
```

Comment:
grammar. "Look for both the" (missing the 'the')

Reply:
grammar. "in the user utterance". (missing the 'the')

Reply:
spelling. uttrance != utterance.

Reply:
Fixed.